tests/system/python/conftest.py

```diff
@@ -410,6 +410,294 @@ def _read_data_from_pi(host, admin, password, pi_database, asset, sensor):
+    url_recorded_data = None
+    # Resources in the PI Web API are addressed by WebID, parameter used for deletion of element
+    web_id = None
+    # List of elements
+    url_elements_data_list = None
+
+    username_password = "{}:{}".format(admin, password)
+    username_password_b64 = base64.b64encode(username_password.encode('ascii')).decode("ascii")
+    headers = {'Authorization': 'Basic %s' % username_password_b64}
+
+    try:
+        conn = http.client.HTTPSConnection(host, context=ssl._create_unverified_context())
+        conn.request("GET", '/piwebapi/assetservers', headers=headers)
+        res = conn.getresponse()
+        r = json.loads(res.read().decode())
+        dbs = r["Items"][0]["Links"]["Databases"]
+
+        if dbs is not None:
+            conn.request("GET", dbs, headers=headers)
+            res = conn.getresponse()
+            r = json.loads(res.read().decode())
+            for el in r["Items"]:
+                if el["Name"] == pi_database:
+                    url_elements_list = el["Links"]["Elements"]
+
+        af_level_count = 0
+        for level in af_hierarchy_list[:-1]:
+            if url_elements_list is not None:
+                conn.request("GET", url_elements_list, headers=headers)
+                res = conn.getresponse()
+                r = json.loads(res.read().decode())
+                for el in r["Items"]:
+                    if el["Name"] == af_hierarchy_list[af_level_count]:
+                        url_elements_list = el["Links"]["Elements"]
+                        if af_level_count == 0:
+                            web_id_root = el["WebId"]
+                        af_level_count = af_level_count + 1
+
+        conn.request("DELETE", '/piwebapi/elements/{}'.format(web_id_root), headers=headers)
+        r = conn.getresponse()
+        assert r.status == 204, "Could not delete element hierarchy of {}".format(af_hierarchy_list)
+        conn.close()
+
+    except Exception as er:
+        print("Could not delete hierarchy of {} due to {}".format(af_hierarchy_list, er))
+        assert False, "Could not delete hierarchy of {} due to {}".format(af_hierarchy_list, er)
+
+
+def clear_cache(host, admin, password, pi_database):
+    """Method that deletes cache by supplying 'Cache-Control': 'no-cache' in header of GET request for
+        element list.
+    """
+    username_password = "{}:{}".format(admin, password)
+    username_password_b64 = base64.b64encode(username_password.encode('ascii')).decode("ascii")
+    headers = {'Authorization': 'Basic %s' % username_password_b64, 'Cache-Control': 'no-cache'}
+    normal_header = {'Authorization': 'Basic %s' % username_password_b64}
+    try:
+        conn = http.client.HTTPSConnection(host, context=ssl._create_unverified_context())
+        conn.request("GET", '/piwebapi/assetservers', headers=normal_header)
+        res = conn.getresponse()
+        assert res.status == 200, "Could not request asset server of Pi Web API."
```

Comment:
Minor stuff, but PI is always capitalized and this product is written as PI Web API.

Reply:
Yes will fix in new pull request.